src/recv.rs

```diff
@@ -28,7 +28,8 @@ impl<const N: usize> ReceiveBuffer<N> {
 
     /// Returns the number of available bytes remaining in the buffer.
     pub fn available(&self) -> usize {
-        N - self.offset - self.pending.values().fold(0, |acc, data| acc + data.len())
+        N.saturating_sub(self.offset)
+            .saturating_sub(self.pending.values().fold(0, |acc, data| acc + data.len()))
```

Comment:
```suggestion
        N  + self.offset - self.pending.values().fold(0, |acc, data| acc + data.len())
```

Try this I believe the issue is, data is added to pending, then when the data is read offset increases which causes the overflow

So we can either remove offset or add offset to N

Reply:
Remember ``N.saturating_sub(self.offset)`` should be add not sub, but we probably shouldn't use saturing since it will hide bugs in the underlying logic